worldedit-core/src/main/java/com/sk89q/worldedit/command/tool/BrushTool.java

```diff
@@ -219,6 +226,7 @@ public boolean actPrimary(Platform server, LocalConfiguration config, Player pla
             }
 
             try {
+                clearPreview();
                 brush.build(editSession, target.toVector().toBlockPoint(), material, size);
```

Comment:
Here, you clear the preview after you receive the `target`. In that case, won't the brush build the structure in front of the preview, instead of where the review is, since the target is pointing at the preview block?